banjo/parse-type.cpp

```diff
@@ -146,16 +146,27 @@ Parser::postfix_type()
 
 // Parse an array of slice type.
 Type&
- Parser::array_type(Type& t)
- {
-   match(lbracket_tok);
-   if (match_if(rbracket_tok))
-      return on_slice_type(t);
-   Expr& e = expression();
-   match(rbracket_tok);
-   return on_array_type(t, e);
- }
+Parser::array_type(Type& t)
+{
+ match(lbracket_tok);
+ if (match_if(rbracket_tok))
+    return on_slice_type(t);
+ Expr& e = expression();
+ match(rbracket_tok);
+ return on_array_type(t, e);
+}
+
  
+Type&
+Parser::tuple_type()
+{
+ match(lbrace_tok);
+ //FIXME: Needs to be a list of types? Not Expr
```

Comment:
Yes. Fortunately, this grammar is shared with function-type, so you can just call type_list.